src/NHibernate/Engine/TableGroupJoinHelper.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NHibernate.Persister.Entity;
+using NHibernate.SqlCommand;
+
+namespace NHibernate.Engine
+{
+	//Generates table group join if neccessary. Example of generated query with table group join:
+	// SELECT *
+	// FROM Person person0_
+	// INNER JOIN (
+	// IndividualCustomer individual1_ 
+	// INNER JOIN Customer individual1_1_ ON individual1_.IndividualCustomerID = individual1_1_.Id
+	// ) ON person0_.Id = individual1_.PersonID AND individual1_1_.Deleted = @p0
+	internal class TableGroupJoinHelper
+	{
+		internal static bool ProcessAsTableGroupJoin(IReadOnlyList<IJoin> tableGroupJoinables, SqlString[] withClauseFragments, bool includeAllSubclassJoins, JoinFragment joinFragment, Func<string, bool> isSubclassIncluded, ISessionFactoryImplementor sessionFactoryImplementor)
+		{
+			if (!TableGroupJoinHelper.NeedsTableGroupJoin(tableGroupJoinables, withClauseFragments, includeAllSubclassJoins))
```

Comment:
Nit: uneeded class qualifier here.
```suggestion
			if (!NeedsTableGroupJoin(tableGroupJoinables, withClauseFragments, includeAllSubclassJoins))
```